middlewares/authorize.js

```diff
@@ -0,0 +1,12 @@
+const authorizeByRole = function(req, res, next){
+    //req.decoded = { id, email, iat, dst}
+    if(req.decoded.role === 'user' && req.decoded.id === +req.params.id){
+        next()
```

Comment:
ini harusnya get one todo dlu mas trus di cek apakah todo.UserId sama dengan id user yang lagi login, trus dihandle juga misalnya todonya gk ketemu langsung di throw data not found